src/bitcoin-cli.cpp

```diff
@@ -485,56 +512,54 @@ static int CommandLineRPC(int argc, char *argv[])
             args.erase(args.begin()); // Remove trailing method name from arguments vector
         }
-
-        // Execute and handle connection failures with -rpcwait
-        const bool fWait = gArgs.GetBoolArg("-rpcwait", false);
-        do {
-            try {
-                const UniValue reply = CallRPC(rh.get(), method, args);
-
-                // Parse reply
-                const UniValue& result = find_value(reply, "result");
-                const UniValue& error  = find_value(reply, "error");
-
-                if (!error.isNull()) {
-                    // Error
-                    int code = error["code"].get_int();
-                    if (fWait && code == RPC_IN_WARMUP)
-                        throw CConnectionFailed("server in warmup");
-                    strPrint = "error: " + error.write();
-                    nRet = abs(code);
-                    if (error.isObject())
-                    {
-                        UniValue errCode = find_value(error, "code");
-                        UniValue errMsg  = find_value(error, "message");
-                        strPrint = errCode.isNull() ? "" : "error code: "+errCode.getValStr()+"\n";
-
-                        if (errMsg.isStr())
-                            strPrint += "error message:\n"+errMsg.get_str();
-
-                        if (errCode.isNum() && errCode.get_int() == RPC_WALLET_NOT_SPECIFIED) {
-                            strPrint += "\nTry adding \"-rpcwallet=<filename>\" option to bitcoin-cli command line.";
-                        }
+        const UniValue reply = ConnectAndCallRPC(rh.get(), method, args);
+
+        // Parse reply
+        UniValue result = find_value(reply, "result");
+        const UniValue& error = find_value(reply, "error");
+        if (!error.isNull()) {
+            // Error
+            strPrint = "error: " + error.write();
+            nRet = abs(error["code"].get_int());
+            if (error.isObject()) {
+                const UniValue& errCode = find_value(error, "code");
+                const UniValue& errMsg  = find_value(error, "message");
+                strPrint = errCode.isNull() ? "" : ("error code: " + errCode.getValStr() + "\n");
+
+                if (errMsg.isStr()) {
+                    strPrint += ("error message:\n" + errMsg.get_str());
+                }
+                if (errCode.isNum() && errCode.get_int() == RPC_WALLET_NOT_SPECIFIED) {
+                    strPrint += "\nTry adding \"-rpcwallet=<filename>\" option to bitcoin-cli command line.";
+                }
+            }
+        } else {
+            // Display mine.trusted balances for all loaded wallets if -getinfo
+            // was called without -rpcwallet and more than one wallet is loaded.
+            if (gArgs.GetBoolArg("-getinfo", false) && gArgs.GetArgs("-rpcwallet").empty()) {
+                rh.reset(new DefaultRequestHandler());
+                const UniValue listwallets = ConnectAndCallRPC(rh.get(), "listwallets", args);
```

Comment:
`args` is always going to be empty here (see L239). Can you change this to an empty vector? (same for `ConnectAndCallRPRC()` call below)

Reply:
Done.